src/services/GraphsDFG.jl

```diff
@@ -0,0 +1,432 @@
+    return factor
+end
+
+"""
+    $(SIGNATURES)
+Delete a DFGVariable from the DFG using its label.
+"""
+function deleteVariable!(dfg::GraphsDFG, label::Symbol)::DFGVariable
+    if !haskey(dfg.labelDict, label)
+        error("Variable label '$(label)' does not exist in the factor graph")
+    end
+    @error "Delete is not supported in Graphs.jl implementation at present"
+end
+
+#Alias
+"""
+    $(SIGNATURES)
+Delete a referenced DFGVariable from the DFG.
+"""
+deleteVariable!(dfg::GraphsDFG, variable::DFGVariable)::DFGVariable = deleteVariable(dfg, variable.label)
+
+"""
+    $(SIGNATURES)
+Delete a DFGFactor from the DFG using its label.
+"""
+function deleteFactor!(dfg::GraphsDFG, label::Symbol)::DFGFactor
+    if !haskey(dfg.labelDict, label)
+        error("Factor label '$(label)' does not exist in the factor graph")
+    end
+    @error "Delete is not supported in Graphs.jl implementation at present"
+end
+
+# Alias
+"""
+    $(SIGNATURES)
+Delete the referened DFGFactor from the DFG.
+"""
+deleteFactor!(dfg::GraphsDFG, factor::DFGFactor)::DFGFactor = deleteFactor(dfg, factor.label)
+
+# # Returns a flat vector of the vertices, keyed by ID.
+# # Assuming only variables here for now - think maybe not, should be variables+factors?
+"""
+    $(SIGNATURES)
+List the DFGVariables in the DFG.
+Optionally specify a label regular expression to retrieves a subset of the variables.
+"""
+function ls(dfg::GraphsDFG, regexFilter::Union{Nothing, Regex}=nothing)::Vector{DFGVariable}
+    variables = map(v -> v.dfgNode, filter(n -> typeof(n.dfgNode) == DFGVariable, collect(values(dfg.g.vertices))))
+    if regexFilter != nothing
+        variables = filter(v -> occursin(regexFilter, String(v.label)), variables)
+    end
+    return variables
+end
+
+# Alias
+"""
+    $(SIGNATURES)
+List the DFGVariables in the DFG.
+Optionally specify a label regular expression to retrieves a subset of the variables.
+"""
+getVariables(dfg::GraphsDFG, regexFilter::Union{Nothing, Regex}=nothing)::Vector{DFGVariable} = ls(dfg, regexFilter)
+
+"""
+    $(SIGNATURES)
+List the DFGFactors in the DFG.
+Optionally specify a label regular expression to retrieves a subset of the factors.
+"""
+function lsf(dfg::GraphsDFG, regexFilter::Union{Nothing, Regex}=nothing)::Vector{DFGFactor}
+    factors = map(v -> v.dfgNode, filter(n -> typeof(n.dfgNode) == DFGFactor, collect(values(dfg.g.vertices))))
+    if regexFilter != nothing
+        factors = filter(f -> occursin(regexFilter, String(f.label)), factors)
+    end
+    return factors
+end
+
+# Alias
+"""
+    $(SIGNATURES)
+List the DFGFactors in the DFG.
+Optionally specify a label regular expression to retrieves a subset of the factors.
+"""
+getFactors(dfg::GraphsDFG, regexFilter::Union{Nothing, Regex}=nothing)::Vector{DFGFactor} = lsf(dfg, regexFilter)
+
+"""
+    $(SIGNATURES)
+Checks if the graph is fully connected, returns true if so.
+"""
+function isFullyConnected(dfg::GraphsDFG)::Bool
```

Comment:
We should remember to add a ‘hasOrphans’ alias

Reply:
Done!